Libraries/LibCore/ProcessStatisticsReader.cpp

```diff
@@ -103,47 +103,31 @@ HashMap<pid_t, Core::ProcessStatistics> ProcessStatisticsReader::get_all()
 
         // and synthetic data last
         process.username = username_from_uid(process.uid);
+        process.working_directory = get_current_working_directory(process.pid);
+        process.executable_path = get_executable_path(process.pid);
```

Comment:
This seems like an aggressive amount of extra work to do for clients that don't need the information.
The change in resource graph applet pledges below are necessary because you now visit every accessible `/proc/PID` directory and read some links.

The resource graphs are updated every second, so this is a whole bunch of extra I/O for no reason in those processes.

To avoid this, how about we have `working_directory()` and `executable_path()` functions instead, and we can fetch the information for clients who actually need it?

Reply:
The resource graphs don't need a majority of ```ProcessStatisticsReader::get_all()``` data, only memory and CPU stats.
Why load and parse a lot of data only to result in 2 numbers? Maybe ```ProcessStatisticsReader::get_performance_stats()``` or something like this?